EIPS/eip-4844.md

```diff
@@ -0,0 +1,558 @@
+without imposing extra development burdens on rollups to take advantage of this relief.
+Today, rollups use calldata. In the future, rollups will have no choice but to use sharded data (also called "blobs")
+because sharded data will be much cheaper.
+Hence, rollups cannot avoid making a large upgrade to how they process data at least once along the way.
+But what we _can_ do is ensure that rollups need to _only_ upgrade once.
+This immediately implies that there are exactly two possibilities for a stopgap: (i) reducing the gas costs of existing calldata,
+and (ii) bringing forward the format that will be used for sharded data, but not yet actually sharding it.
+Previous EIPs were all a solution of category (i); this EIP is a solution of category (ii).
+
+The main tradeoff in designing this EIP is that of implementing more now versus having to implement more later:
+do we implement 25% of the work on the way to full sharding, or 50%, or 75%?
+
+The work that is already done in this EIP includes:
+
+* A new transaction type, of the exact same format that will need to exist in "full sharding"
+* _All_ of the execution-layer logic required for full sharding
+* _All_ of the execution / consensus cross-verification logic required for full sharding
+* Layer separation between `BeaconBlock` verification and data availability sampling blobs
+* Most of the `BeaconBlock` logic required for full sharding
+* A self-adjusting independent gasprice for blobs.
+
+The work that remains to be done to get to full sharding includes:
+
+* A low-degree extension of the `blob_kzgs` in the consensus layer to allow 2D sampling
+* An actual implementation of data availability sampling
+* PBS (proposer/builder separation), to avoid requiring individual validators to process 32 MB of data in one slot
+* Proof of custody or similar in-protocol requirement for each validator to verify a particular part of the sharded data in each block
+
+This EIP also sets the stage for longer-term protocol cleanups:
+
+* It adds an SSZ transaction type which is slightly gas-advantaged (1000 discount) to nudge people toward using it,
+  and paves the precedent that all new transaction types should be SSZ
+* It defines `TransactionNetworkPayload` to separate network and block encodings of a transaction type
+* Its (cleaner) gas price update rule could be applied to the primary basefee.
+  It also starts the trend toward using the state to store system state instead of having "implied state" such as historical blocks and hashes.
+
+### How rollups would function
+
+Instead of putting rollup block data in transaction calldata, rollups would expect rollup block submitters
+to put the data into blobs. This guarantees availability (which is what rollups need) but would be much cheaper than calldata.
+Rollups need data to be available once, long enough to ensure honest actors can construct the rollup state, but not forever.
```

Comment:
Thus why we can have different p2p/node pruning rules than the rest of block data